From 9bd7df848d7e683eb78637286a00af1c86dee361 Mon Sep 17 00:00:00 2001
From: Peter Huewe <peterhuewe@gmx.de>
Date: Thu, 14 Jun 2018 22:42:18 +0200
Subject: [PATCH 448/454] Enable TPM TIS SPI support for TPM1.2 and TPM2.0
 chips

This patch enables the support for SPI TPMs which follow the TCG TIS
FIFO/PTP specification like the SLB9670.
In order to decrease ram usage the weak dependency on CONFIG_SECURITFS
is explictly set to 'n'.

Signed-off-by: Peter Huewe <peterhuewe@gmx.de>
---
 arch/arm/configs/bcm2709_defconfig   |  3 +++
 arch/arm/configs/bcmrpi_defconfig    |  3 +++
 arch/arm64/configs/bcmrpi3_defconfig | 11 ++++++-----
 3 files changed, 12 insertions(+), 5 deletions(-)

--- a/arch/arm/configs/bcm2709_defconfig
+++ b/arch/arm/configs/bcm2709_defconfig
@@ -607,6 +607,8 @@ CONFIG_SERIAL_DEV_BUS=m
 CONFIG_TTY_PRINTK=y
 CONFIG_HW_RANDOM=y
 CONFIG_RAW_DRIVER=y
+CONFIG_TCG_TPM=m
+CONFIG_TCG_TIS_SPI=m
 CONFIG_I2C=y
 CONFIG_I2C_CHARDEV=m
 CONFIG_I2C_MUX_GPMUX=m
@@ -1361,6 +1363,7 @@ CONFIG_FUNCTION_PROFILER=y
 CONFIG_KGDB=y
 CONFIG_KGDB_KDB=y
 CONFIG_KDB_KEYBOARD=y
+# CONFIG_SECURITYFS is not set
 CONFIG_CRYPTO_USER=m
 CONFIG_CRYPTO_CBC=y
 CONFIG_CRYPTO_CTS=m
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -602,6 +602,8 @@ CONFIG_SERIAL_DEV_BUS=m
 CONFIG_TTY_PRINTK=y
 CONFIG_HW_RANDOM=y
 CONFIG_RAW_DRIVER=y
+CONFIG_TCG_TPM=m
+CONFIG_TCG_TIS_SPI=m
 CONFIG_I2C=y
 CONFIG_I2C_CHARDEV=m
 CONFIG_I2C_MUX_GPMUX=m
@@ -1353,6 +1355,7 @@ CONFIG_FUNCTION_PROFILER=y
 CONFIG_KGDB=y
 CONFIG_KGDB_KDB=y
 CONFIG_KDB_KEYBOARD=y
+# CONFIG_SECURITYFS is not set
 CONFIG_CRYPTO_USER=m
 CONFIG_CRYPTO_CRYPTD=m
 CONFIG_CRYPTO_CBC=y
--- a/arch/arm64/configs/bcmrpi3_defconfig
+++ b/arch/arm64/configs/bcmrpi3_defconfig
@@ -98,6 +98,8 @@ CONFIG_INET_XFRM_MODE_TRANSPORT=m
 CONFIG_INET_XFRM_MODE_TUNNEL=m
 CONFIG_INET_XFRM_MODE_BEET=m
 CONFIG_INET_DIAG=m
+CONFIG_TCP_CONG_ADVANCED=y
+CONFIG_TCP_CONG_BBR=m
 CONFIG_IPV6=m
 CONFIG_IPV6_ROUTER_PREF=y
 CONFIG_INET6_AH=m
@@ -306,11 +308,9 @@ CONFIG_NET_SCH_CHOKE=m
 CONFIG_NET_SCH_QFQ=m
 CONFIG_NET_SCH_CODEL=m
 CONFIG_NET_SCH_FQ_CODEL=m
+CONFIG_NET_SCH_FQ=m
 CONFIG_NET_SCH_INGRESS=m
 CONFIG_NET_SCH_PLUG=m
-CONFIG_NET_SCH_FQ=m
-CONFIG_TCP_CONG_ADVANCED=y
-CONFIG_TCP_CONG_BBR=m
 CONFIG_NET_CLS_BASIC=m
 CONFIG_NET_CLS_TCINDEX=m
 CONFIG_NET_CLS_ROUTE4=m
@@ -586,6 +586,8 @@ CONFIG_SERIAL_DEV_BUS=m
 CONFIG_TTY_PRINTK=y
 CONFIG_HW_RANDOM=y
 CONFIG_RAW_DRIVER=y
+CONFIG_TCG_TPM=m
+CONFIG_TCG_TIS_SPI=m
 CONFIG_I2C=y
 CONFIG_I2C_CHARDEV=m
 CONFIG_I2C_MUX_PCA954x=m
@@ -1265,6 +1267,7 @@ CONFIG_FUNCTION_PROFILER=y
 CONFIG_KGDB=y
 CONFIG_KGDB_KDB=y
 CONFIG_KDB_KEYBOARD=y
+# CONFIG_SECURITYFS is not set
 CONFIG_CRYPTO_USER=m
 CONFIG_CRYPTO_CBC=y
 CONFIG_CRYPTO_CTS=m
@@ -1277,8 +1280,6 @@ CONFIG_CRYPTO_DES=y
 CONFIG_CRYPTO_LZ4=m
 CONFIG_CRYPTO_USER_API_SKCIPHER=m
 CONFIG_ARM64_CRYPTO=y
-CONFIG_CRYPTO_AES_ARM64=m
 CONFIG_CRYPTO_AES_ARM64_BS=m
-CONFIG_CRYPTO_AES_ARM64_NEON_BLK=m
 CONFIG_CRC_ITU_T=y
 CONFIG_LIBCRC32C=y
